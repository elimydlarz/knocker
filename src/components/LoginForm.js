import React, { Component } from 'react';
import TextInput from './TextInput';
import Submit from './Submit';
import '../css/LoginForm.css';

export default class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.loginSuccess = this.loginSuccess.bind(this);
    this.loginFailure = this.loginFailure.bind(this);
  }

  loginSuccess() {
    this.props.updateMessage(`Welcome to Knocker, ${this.props.user}!`);
    this.props.updateAuthorised(true);
  }

  loginFailure(response) {
    const message = response.status === 401
      ? `We're having trouble authenticating you, ${this.props.user}.
        Make sure you've identified yourself correctly and
        supplied the right password.`
      : 'Whoops! Something unexpected has gone wrong.'

    this.props.updateMessage(message);
  }

  render() {
    return (
      <div className='Login'>
        <form>
          <TextInput
            name='user'
            label="Who's knocking?"
            value={this.props.user}
            changeHandler={this.props.userHandler}
          />
          <input type='password' name='password'
            placeholder="What's your password?"
            value={this.props.password}
            onChange={this.props.passwordHandler}
          />
          <Submit
            user={this.props.user}
            password={this.props.password}
            eventType='login'
            data={{ user: this.props.user }}
            requiredFields={[
              this.props.user,
              this.props.password,
            ]}
            successFn={this.loginSuccess}
            failureFn={this.loginFailure}
          />
        </form>
      </div>
    );
  }
}
