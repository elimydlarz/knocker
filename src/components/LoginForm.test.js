import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';

describe('LoginForm', () => {
  it('does not explode', () => {
    const component = renderer.create(
      <LoginForm
        user='COOL USER'
        password='COOL PASSWORD'
        userHandler={() => {}}
        passwordHandler={() => {}}
        updateMessage={() => {}}
        updateAuthorised={() => {}}
      />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
