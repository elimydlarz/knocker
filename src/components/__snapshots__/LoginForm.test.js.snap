exports[`LoginForm does not explode 1`] = `
<div
  className="Login">
  <form>
    <div
      className="TextInput">
      <input
        name="user"
        onChange={[Function userHandler]}
        placeholder="Who\'s knocking?"
        type="text"
        value="COOL USER" />
    </div>
    <input
      name="password"
      onChange={[Function passwordHandler]}
      placeholder="What\'s your password?"
      type="password"
      value="COOL PASSWORD" />
    <div
      className="Submit">
      <input
        disabled={false}
        onClick={[Function bound transmit]}
        type="submit" />
    </div>
  </form>
</div>
`;
